### PR TITLE
Added serialization of parameters for leaf modules

### PR DESCRIPTION
**Pull request by gcatron**
This adds the serialization of parameters of leaf nodes to the json serialization.
Specifically __constants__ of the leaf module is serialized as parameters in the JSON.
It also adds type/shape to leaf modules as well.
```
{
            "shape": "[3, 3, 1, 1]",
            "dtype": "torch.float32",
            "parameters": {
                "name": "Conv2d",
                "stride": [
                    1,
                    1
                ],
                "padding": [
                    0,
                    0
                ],
                "dilation": [
                    1,
                    1
                ],
                "groups": 1,
                "padding_mode": "zeros",
                "output_padding": [
                    0,
                    0
                ],
                "in_channels": 3,
                "out_channels": 3,
                "kernel_size": [
                    2,
                    2
                ]
            },
            "target": "conv",
            "op_code": "call_module",
            "name": "conv",
            "args": [
                {
                    "is_node": true,
                    "name": "c"
                }
            ],
            "kwargs": {}
        },
```

### CODE REVIEW

File: torch/fx/experimental/GraphManipulation.py
Comment (842974287) [inline torch/fx/experimental/GraphManipulation.py:135]: Would it be possible to make the value a string, like `str(getattr(mod, constant))`? This would help with the processing because constant like `kernel_size` could be an integer or a tuple.

**Review by 842974287 — COMMENTED** [review]
Looks pretty good!

**Review by 842974287 — APPROVED** [review]

File: torch/fx/experimental/GraphManipulation.py
Comment (jamesr66a) [inline torch/fx/experimental/GraphManipulation.py:135]: OK, I understand what's going on now. `__constants__` is here because the standard `nn.Module` instances just happen to have those defined for TorchScript compatibility. This is *not* a part of the `nn.Module` API, thus this will break in the general case. My suggestion is to write functions for each `nn.Module` type to pull out the pertinent attributes for lowering

**Review by jamesr66a — COMMENTED** [review]
Likely should not land this

File: torch/fx/experimental/GraphManipulation.py
Comment (jamesr66a) [reply torch/fx/experimental/GraphManipulation.py:135]: And for more constants, these field names are listed in `__constants__` because they're used in `forward()`, there is no contract that says that `__constants__` contains all of the pertinent information for lowering this `Module`

Comment (gcatron) [reply torch/fx/experimental/GraphManipulation.py:135]: Good catch, per our offline discussion I'll go ahead and land this for now to unblock testing but @842974287 will follow up with a more complete paramater harvesting pass and we'll update the JSON serialization then.

**Pull request closed without merging**
